api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -436,12 +497,12 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 		summary.SuccessCount = summary.TargetCount
 	}
 
-	s.saveSummary(iCtx, deployment, summary, namespace)
+	s.concludeSummary(iCtx, deployment, summary, namespace)
```

Comment:
Do we need this? We have concludeSummary in defer function. 
Besides, saveSummaryProgress saves summary and set state to InProgress. I am curious why we need to set it at Line 431

Reply:
Good catch. This is a code rebase issue. We keep conclude summary in defer function. There's no need to call it explicitly. 
L431 is to report progress per step, which allows the controller to get the progress information and reflect them e.g. in ARM.